debug_toolbar/panels/redirects.py

```diff
@@ -9,23 +11,34 @@ class RedirectsPanel(Panel):
     Panel that intercepts redirects and displays a page with debug info.
     """
 
-    is_async = False
+    is_async = True
     has_content = False
 
     nav_title = _("Intercept redirects")
 
-    def process_request(self, request):
-        response = super().process_request(request)
+    def _process_response(self, response):
+        """
+        Common response processing logic.
+        """
         if 300 <= response.status_code < 400:
             redirect_to = response.get("Location")
             if redirect_to:
                 status_line = f"{response.status_code} {response.reason_phrase}"
                 cookies = response.cookies
                 context = {"redirect_to": redirect_to, "status_line": status_line}
-                # Using SimpleTemplateResponse avoids running global context processors.
                 response = SimpleTemplateResponse(
                     "debug_toolbar/redirect.html", context
                 )
                 response.cookies = cookies
                 response.render()
         return response
+
+    async def aprocess_request(self, request):
+        response = await super().process_request(request)
+        return self._process_response(response)
+
+    def process_request(self, request):
+        response = super().process_request(request)
+        if iscoroutine(response):
+            return self.aprocess_request(request)
```

Comment:
This flow will call super().process_request twice. We probably want to avoid that. I have some ideas, but none that I know are great. What do you think?

Reply:
since `aprocess_request` is exclusive to `RedirectsPanel`
1. we can pass response from `process_request` in `aprocess_request` as an argument.
2. or else we change the check from coroutine to checking request being an instance `ASGIRequest`.

Reply:
My preference is option 1

Reply:
I think 2. won't work either : )